ros_gz_sim/src/gz_simulation_interfaces/services/spawn_entity.cpp

```diff
@@ -0,0 +1,108 @@
+// Copyright 2025 Open Source Robotics Foundation, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include "spawn_entity.hpp"
+
+#include <gz/msgs/boolean.pb.h>
+#include <gz/msgs/entity_factory.pb.h>
+
+#include <memory>
+
+#include "../gazebo_proxy.hpp"
+#include "simulation_interfaces/srv/spawn_entity.hpp"
+
+namespace ros_gz_sim
+{
+namespace gz_simulation_interfaces
+{
+namespace services
+{
+using SpawnEntitySrv = simulation_interfaces::srv::SpawnEntity;
+using RequestPtr = SpawnEntitySrv::Request::ConstSharedPtr;
+using ResponsePtr = SpawnEntitySrv::Response::SharedPtr;
+
+SpawnEntity::SpawnEntity(
+  std::shared_ptr<rclcpp::Node> ros_node, std::shared_ptr<GazeboProxy> gz_proxy)
+: HandlerBase(ros_node, gz_proxy)
+{
+  const auto create_service = this->gz_proxy_->PrefixTopic("create/blocking");
+  if (!this->gz_proxy_->WaitForGzService(create_service)) {
+    RCLCPP_ERROR_STREAM(
+      this->ros_node_->get_logger(),
+      "Gazebo service [" << create_service << "] is not available. "
+                         << "The [SpawnEntity] interface will not function properly. To fix this, "
+                            "make sure the [UserCommands] system is loaded in your Gazebo world");
+  }
+  this->services_handle_ = ros_node->create_service<SpawnEntitySrv>(
+    "spawn_entity", [this, create_service](RequestPtr request, ResponsePtr response) {
+      using Result = simulation_interfaces::msg::Result;
+      gz::msgs::EntityFactory gz_request;
+      if (!request->name.empty()) {
+        gz_request.set_name(request->name);
+      }
+      gz_request.set_allow_renaming(request->allow_renaming);
+      const auto & resource = request->entity_resource;
+
+      if (!resource.uri.empty()) {
+        // TODO(azeey) The `sdf_filename` field requires absolute paths to the file.
+        // Consider resolving the uri using the `/gazebo/resource_paths/resolve`
+        gz_request.set_sdf_filename(resource.uri);
+      } else if (!resource.resource_string.empty()) {
+        gz_request.set_sdf(resource.resource_string);
```

Comment:
The `entity_resource` field is new in `rolling`. For `kilted` we'll just need to use the `uri` and `resource_string` [fields](https://github.com/ros-simulation/simulation_interfaces/blob/c35c28a768eaa875ed4269a8cbc8e4d6ec482eb5/srv/SpawnEntity.srv#L14) directly, so maybe something like:

```suggestion
      if (!request->uri.empty()) {
        // TODO(azeey) The `sdf_filename` field requires absolute paths to the file.
        // Consider resolving the uri using the `/gazebo/resource_paths/resolve`
        gz_request.set_sdf_filename(request->uri);
      } else if (!request->resource_string.empty()) {
        gz_request.set_sdf(request->resource_string);
```